From f54d12d9f69ca268ec94369ef1b0f7e6cdd40aaf Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 7 Dec 2017 16:53:02 +0800
Subject: [PATCH 3005/5242] MLK-17112 gpu: imx: dpu: Ensure the dpu core is
 late suspended & early resumed

commit  e42c9ed32af06899d045b63502c9e9392abb1f26 from
https://source.codeaurora.org/external/imx/linux-imx.git

The dpu core driver needs to do some cleanup work for the
upper layer drivers, so it should be late suspended and
early resumed.  This patch ensures this by changing driver
PM hook definitions to use SET_LATE_SYSTEM_SLEEP_PM_OPS
instead of SET_SYSTEM_SLEEP_PM_OPS.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index 9249edf..8d0ea76 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -1571,7 +1571,9 @@ static int dpu_resume(struct device *dev)
 }
 #endif
 
-static SIMPLE_DEV_PM_OPS(dpu_pm_ops, dpu_suspend, dpu_resume);
+static const struct dev_pm_ops dpu_pm_ops = {
+	SET_LATE_SYSTEM_SLEEP_PM_OPS(dpu_suspend, dpu_resume)
+};
 
 static struct platform_driver dpu_driver = {
 	.driver = {
-- 
1.7.9.5

